content/tokio/tutorial/bridging.md

```diff
@@ -0,0 +1,384 @@
+
+The `#[tokio::main]` macro is a macro that replaces your main function with a
+non-async main function that starts a runtime and then calls your code. For
+instance, this:
+```rust
+#[tokio::main]
+async fn main() {
+    println!("Hello world");
+}
+```
+is turned into this:
+```
+fn main() {
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+            println!("Hello world");
+        })
+}
+```
+by the macro. To use async/await in our own projects, we can do something
+similar where we leverage the [`block_on`] method to enter the asynchronous
+context where appropriate.
+
+# A synchronous interface to mini-redis
+
+In this section, we will go through how to build a synchronous interface to
+mini-redis by storing a `Runtime` object and using its `block_on` method.
+In the following sections, we will discuss some alternate approaches and when
+you should use each approach.
+
+The interface that we will be wrapping is the asynchronous [`Client`] type. It
+has several methods, and we will implement a blocking version of the following
+methods:
+
+ * [`Client::get`]
+ * [`Client::set`]
+ * [`Client::set_expires`]
+ * [`Client::publish`]
+ * [`Client::subscribe`]
+
+To do this, we introduce a new file called `src/blocking_client.rs` and
+initialize it with a wrapper struct around the async `Client` type:
+```rs
+use tokio::net::ToSocketAddrs;
+use tokio::runtime::Runtime;
+
+pub use crate::client::Message;
+
+/// Established connection with a Redis server.
+pub struct Client {
+    /// The asynchronous `Client`.
+    inner: crate::client::Client,
+
+    /// A `current_thread` runtime for executing operations on the
+    /// asynchronous client in a blocking manner.
+    rt: Runtime,
+}
+
+pub fn connect<T: ToSocketAddrs>(addr: T) -> crate::Result<Client> {
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()?;
+
+    // Call the asynchronous connect method using the runtime.
+    let inner = rt.block_on(crate::client::connect(addr))?;
+
+    Ok(Client { inner, rt })
+}
+```
+Here, we have included the constructor function as our first example of how to
+execute asynchronous methods in a non-async context. We do this using the
+[`block_on`] method on the Tokio [`Runtime`] type, which executes an
+asynchronous method and returns its result.
+
+One important detail is the use of the [`current_thread`] runtime. Normally when
+using Tokio, you would be using the default [`multi_thread`] runtime, but this
+runtime will spawn a bunch of background threads so it can efficiently run many
+things at the same time. For our use-case, we are only going to be doing one
+thing at the time, so we only need a light-weight runtime, and the
+[`current_thread`] runtime is perfect for this as it doesn't spawn any threads.
+
+[[warning]]
+| Since the `current_thread` runtime doesn't spawn any threads, it is not
```

Comment:
The warning is a bit confusing. The initial sentence reads like you can't spawn tasks in the background but then you say you can. Maybe something like:

> Because the `current_thread` runtime does not spawn threads, it only operates when `block_on` is called. Once `block_on` returns, all spawned tasks on that runtime will freeze until you call `block_on` again. Use the `multi_threaded` runtime if spawned tasks must keep running when **not** calling `block_on`.